Sources/Controllers/2.0/BottomSheet/SettingsActionBottomSheet/OACopyProfileBottomSheetViewController.mm

```diff
@@ -0,0 +1,620 @@
+        case EOACopyProfileMenuStateFullScreen:
+            return DeviceScreenHeight - OAUtilities.getTopMargin;
+        default:
+            return 0.0;
+    }
+}
+
+- (void) show:(BOOL)animated
+{
+    [_tableView setContentOffset:CGPointZero];
+    _isHiding = NO;
+    _currentState = OAUtilities.isLandscape ? EOACopyProfileMenuStateFullScreen : EOACopyProfileMenuStateInitial;
+    [_tableView setScrollEnabled:_currentState == EOACopyProfileMenuStateFullScreen];
+    [self generateData];
+    [self setNeedsLayout];
+    [self adjustFrame];
+    [self.tableView reloadData];
+    if (animated)
+    {
+        CGRect frame = self.frame;
+        if (OAUtilities.isLandscape)
+        {
+            frame.origin.x = DeviceScreenWidth/2 - frame.size.width / 2;
+            frame.size.width = OAUtilities.isIPad ? kOABottomSheetWidthIPad : kOABottomSheetWidth;
+        }
+        else
+        {
+            frame.origin.x = 0.0;
+            frame.size.width = DeviceScreenWidth;
+        }
+        frame.origin.y = DeviceScreenHeight + 10;
+        self.frame = frame;
+        frame.origin.y = DeviceScreenHeight - self.bounds.size.height;
+        [UIView animateWithDuration:.3 animations:^{
+            self.frame = frame;
+        }];
+    }
+    else
+    {
+        CGRect frame = self.frame;
+        if (OAUtilities.isLandscape)
+            frame.origin.y = 0.0;
+        else
+            frame.origin.y = DeviceScreenHeight - self.bounds.size.height;
+        self.frame = frame;
+    }
+}
+
+- (void) hide:(BOOL)animated
+{
+    _isHiding = YES;
+    
+    if (self.superview)
+    {
+        CGRect frame = self.frame;
+        frame.origin.y = DeviceScreenHeight + 10.0;
+        
+        if (animated)
+        {
+            [UIView animateWithDuration:0.3 animations:^{
+                self.frame = frame;
+            } completion:^(BOOL finished) {
+                if (self.delegate)
+                    [self.delegate onCopyProfileDismessed];
+                [self removeFromSuperview];
+            }];
+        }
+        else
+        {
+            self.frame = frame;
+            [self removeFromSuperview];
+        }
+    }
+}
+
+# pragma mark - Buttons Actions
+
+- (IBAction) closeButtonPressed:(id)sender
+{
+    [self hide:YES];
+}
+
+- (IBAction) cancelButtonPressed:(id)sender
+{
+    [self hide:YES];
+}
+
+- (IBAction) copyProfileButtonPressed:(id)sender
+{
+    [self copyProfile];
+    if (self.delegate)
+        [self.delegate onCopyProfileCompleted];
+    [self hide:YES];
+}
+
+- (void) copyRegisteredPreferences
+{
+    for (NSString *key in _settings.getRegisteredSettings)
+    {
+        OAProfileSetting *setting = [_settings.getRegisteredSettings objectForKey:key];
+        if (setting)
+        {
+            if ([key isEqualToString:@"voice_provider"])
+                [setting setValueFromString:[[setting toStringValue:_sourceAppMode] stringByReplacingOccurrencesOfString:@"-tts" withString:@""] appMode:_targetAppMode];
```

Comment:
This is copied from SettingsHelper. You do not need to use setValueFromString and setValueFromString at all here. This code was meant to synchronize values with Android. For internal settings, you simply need to use ```[setting set:[setting get:_sourceAppMode] mode:_targetAppMode]